[quant][graphmode][refactor] Factor out insertDequantCall

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33650 [not4land] debugging
* #33533 [not4land] Temporary workaround to get quantized ops
* #33481 [quant][graphmode] Handling ops doesn't require observation in insertObservers
* #33532 [quant][graphmode] Add quantization pattern for quantized::add_relu
* #33279 [quant][graphmode] Add quantized conv2d-relu fusion pattern
* #32816 [quant][graphmode] Add add_relu pattern in skip values
* #32814 [quant][graphmode] Skip quantizing input and output in matched module
* #33277 [WIP][quant][graphmode] Observing input/output values in callsite
* #33649 [quant][graphmode][refactor] Factor out getInvokedMethod
* #33276 [quant][graphmode][refactor] Move check for weight outside of insertObserverFor
* #33275 [quant][graphmode][refactor] Move values_to_skip check inside valueNeedsToBeQuantized
* #33274 [quant][graphmode][refactor] Simplify signature for insertObserverFor
* #33273 [quant][graphmode][refactor] Checks for bias and weight
* #33173 [quant][graphmode] Add a pass to handle ops that doesn't require observation
* #33531 [quant][graphmode] Replicate dequantize nodes
* **#33172 [quant][graphmode][refactor] Factor out insertDequantCall**
* #33171 [quant][graphmode] refactor nodeQuantizable

Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20087842](https://our.internmc.facebook.com/intern/diff/D20087842)

jerryzh168 committed:
[quant][graphmode][refactor] Factor out insertDequantCall

Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

dreiss commented on torch/csrc/jit/passes/quantization.cpp:
Looks like this is supposed to be part of the previous diff.  Please split/merge to make sure bisecting works.

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out insertDequantCall"


Summary:
For code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20087842](https://our.internmc.facebook.com/intern/diff/D20087842)

[ghstack-poisoned]
